collections/binary_search.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+// This module is browser compatible.
+
+/**
+ * A binary search that accounts for non-exact matches.
```

Comment:
Can you mention that the array needs to be sorted in the first sentence? (Maybe borrowing some expressions from C#'s or Java's?)

Reply:
Also it'd be better to highlight when this should be preferred over `array.findIndex()`